Archive/PiBase/Properties.lean

```diff
@@ -0,0 +1,25 @@
+import Mathlib.Topology.Basic
+import Mathlib.Topology.Separation
+import Mathlib.Order.Filter.Basic
+import Mathlib.Logic.Nontrivial.Basic
+
+open Filter Topology Nontrivial
+
+universe u
+
+namespace πBase
+
+def P1 (X : Type u) [TopologicalSpace X] := T0Space X
+
+def P2 (X : Type u) [TopologicalSpace X] := T1Space X
+
+def P78 (X : Type u) [TopologicalSpace X] := Finite X
+
+def P99 (X : Type u) [TopologicalSpace X] :=
+  ∀ x y : X, ∀ f : ℕ → X,
+    Tendsto f atTop (𝓝 x) → Tendsto f atTop (𝓝 y) →
+    x = y
+
+def P125 (X : Type u) [TopologicalSpace X] := Nontrivial X
```

Comment:
I think making all of these defs as opposed to abrevs will hide them from the typeclass system, making them harder to work with. Those of these that refer to classes and not just structures or defs (i.e T0Space T1Space, Finite, Nontrivial) should almost certainly be changed to abrev, and maybe they all should in fact

Reply:
Thanks for helping with this subtlety. @JADekker suggested something similar; I'm not seeing any change in behavior when using abbrev (and don't fully understand the difference), and want to make sure that this proof of concept is modeling what I eventually want to produce from pi-Base data algoirthmically (once that data is enhanced with appropriate Lean code).

Reply:
Sure the difference is whether or not various components of the system can "see through" the definition. https://proofassistants.stackexchange.com/questions/2457/what-is-the-precise-meaning-of-reducible has a good summary. Often in mathematics and in formalization we want to hide the precise details of the definition, in lean this from being unfolded all the time and the system getting bogged down by details. But in this case you are using these names as like tags or essentially just your own names for some existing concepts, so everything will behave more how you expect with abbrev